### ISSUE TITLE
fix LossFunction and ActivationFunction class structure

### ISSUE THREAD

**Issue by MirkoCalvi** [ZIGTinyBook/TheTinyBook#59]
Now loss function and activation function must be hardcoded, it's verbose and gives a lot of problems for example with Softmax Backward where I have to use the output of the previous forward of Softamx activation.
